host/host.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+var spawn = require('child_process').spawn;
+var uuid = require('uuid');
+var request = require('./lib/request');
+var assert = require('assert');
+
+var EventEmitter = require('events').EventEmitter;
+var Promise = require('promise');
+
+var VENV = __dirname + '/../venv';
+
+/**
+Wrapper for spawning a python process which expects a venv with particular
+packages. Same interface as spawn but overrides path and ensures certain env
+variables are not set which conflict.
+
+@param {String} bin path to binary to execute.
+@param {Array} argv list of arguments.
+@param {Object} opts options for node's spawn.
+@return {ChildProcess}
+*/
+function spawnVirtualEnv(bin, argv, opts) {
+  opts = opts || {};
+  // Clone current environment variables...
+  var env = opts.env = {};
+  for (var key in process.env) env[key] = process.env[key];
```

Comment:
(not using underscore here though I would like to use 6to5 soon and just use Object.assign)